pkg/3scale/amp/operator/base_apimanager_logic_reconciler.go

```diff
@@ -82,3 +83,11 @@ func (r *BaseAPIManagerLogicReconciler) updateResource(obj common.KubernetesObje
 	r.Logger().Info(fmt.Sprintf("Updated object %s", ObjectInfo(obj)))
 	return r.Client().Update(context.TODO(), obj) // don't wrap error
 }
+
+func (r *BaseAPIManagerLogicReconciler) reconcilePodDisruptionBudget(desiredPDB *v1beta1.PodDisruptionBudget) error {
+	if r.apiManager.Spec.PodDisruptionBudget.Enabled {
```

Comment:
I've been thinking about this.

I'm still thinking on what would be the appropriate way to expose this in the CR. What structure to have and how configurable/what granularity it should have.

If we assume we will use the implemented approach at this moment I have the following comment:

If you deploy an APIManager with this setting enabled it will deploy and reconcile the PodDisruptionBudgets. After APIManager has been deployed, if the user decides to change the PodDisruptionBudget.Enabled to false the PodDisruptionBudget will not be reconciled anymore but it will still exist, and being used in the specified Deployments with the label.
This means this parameter as it is implemented right now it is a "deploy-only" attribute in the CR.

I think we might add reconciliation code to check that when it is not enabled we ensure the PodDisruptionBudgets do not exist, and in case they do remove them. cc @eguzki what do you think?

Reply:
I agree that we could do that, however, I am not sure we should do that, a few reasons:

1. I struggle to find a real world use case where you want to enable PodDisruptionBudgets and then disable it later. Especially with the way that we create those `PodDisruptionBudgets`- they still allow pods to be evicted, just need to be one pod at a time. And scaling up/down pods are not affected at all.
2. It will make code more complicated, for a use case that we don't know if ever going to be needed. On the other hand, we can create docs and explain that if user ever need to remove the PDBs, they can disable it in the CR and manually delete them.

Of course we can consider this again if users come back with good reasons why they need this feature, and then we can implement it. For now, I think we should keep it simple.

Reply:
Hi @wei-lee,

Although we understand your reasons we would like to have the ability to be able to change the field:
 * It would be more intuitive and consistent for the user if he wants to change the value. We try to avoid, if possible/reasonable, fields in the CR that are "non-reconcilable"
 * There might be some case the user / we might want to temporarily disable it. If you want to make a maintenance where you want to go faster (don't mind that several pods are down at the same time), testing of functionality, reduce complexity of the deployment if you realize you don't ned it...

The idea would be that in case the flag is not enabled, if the object exists we would delete it. Unless I'm missing something I think that the functionality might be contained in the reconcilePodDisruptionBudget method.

Reply:
@miguelsorianod fair enough. I will add support for removing the PDBs as you suggested. Thanks.

Reply:
@miguelsorianod I have updated the PR as suggested. Now the operator will remove the PDBs that are created by the operator if the flag is disabled. You see from [this commit](https://github.com/3scale/3scale-operator/pull/308/commits/06cfd21e2d2f24de2f71f08552a0a87820aecfec).

Reply:
looks good to me :+1: